subprojects/diagnostics/src/main/java/org/gradle/api/plugins/ProjectReportsPlugin.java

```diff
@@ -42,31 +41,34 @@ public void apply(final Project project) {
-        taskReportTask.conventionMapping("projects", convention::getProjects);
+        project.getTasks().register(TASK_REPORT, TaskReportTask.class, taskReportTask -> {
+            taskReportTask.setDescription("Generates a report about your tasks.");
+            taskReportTask.conventionMapping("outputFile", () -> new File(convention.getProjectReportDir(), "tasks.txt"));
+            taskReportTask.conventionMapping("projects", convention::getProjects);
+        });
 
-        PropertyReportTask propertyReportTask = project.getTasks().create(PROPERTY_REPORT, PropertyReportTask.class);
-        propertyReportTask.setDescription("Generates a report about your properties.");
-        propertyReportTask.conventionMapping("outputFile", () -> new File(convention.getProjectReportDir(), "properties.txt"));
-        propertyReportTask.conventionMapping("projects", convention::getProjects);
+        project.getTasks().register(PROPERTY_REPORT, PropertyReportTask.class, propertyReportTask -> {
+            propertyReportTask.setDescription("Generates a report about your properties.");
+            propertyReportTask.conventionMapping("outputFile", () -> new File(convention.getProjectReportDir(), "properties.txt"));
+            propertyReportTask.conventionMapping("projects", convention::getProjects);
+        });
 
-        DependencyReportTask dependencyReportTask = project.getTasks().create(DEPENDENCY_REPORT,
-                DependencyReportTask.class);
-        dependencyReportTask.setDescription("Generates a report about your library dependencies.");
-        dependencyReportTask.conventionMapping("outputFile", () -> new File(convention.getProjectReportDir(), "dependencies.txt"));
-        dependencyReportTask.conventionMapping("projects", convention::getProjects);
+        project.getTasks().register(DEPENDENCY_REPORT, DependencyReportTask.class, dependencyReportTask -> {
+            dependencyReportTask.setDescription("Generates a report about your library dependencies.");
+            dependencyReportTask.conventionMapping("outputFile", () -> new File(convention.getProjectReportDir(), "dependencies.txt"));
+            dependencyReportTask.conventionMapping("projects", convention::getProjects);
+        });
 
-        HtmlDependencyReportTask htmlDependencyReportTask = project.getTasks().create(HTML_DEPENDENCY_REPORT,
-                HtmlDependencyReportTask.class);
-        htmlDependencyReportTask.setDescription("Generates an HTML report about your library dependencies.");
-        htmlDependencyReportTask.getReports().getHtml().getOutputLocation().convention(project.getLayout().getProjectDirectory().dir(project.provider(() -> new File(convention.getProjectReportDir(), "dependencies").getAbsolutePath())));
-        htmlDependencyReportTask.conventionMapping("projects", convention::getProjects);
+        project.getTasks().create(HTML_DEPENDENCY_REPORT, HtmlDependencyReportTask.class, htmlDependencyReportTask -> {
```

Comment:
@gaiazov I know I'm late to the party, but do you happen to remember why this task was left to `create` instead of `register`?

Reply:
I think it was by accident 😓